pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+            } else if (message.indexOf(invalidCredentialsString) !== -1) {
+                message = cockpit.format("$0 ($1)", _("Invalid credentials"), message.substring(invalidCredentialsString.length).trim());
+            } else if ((message.indexOf('EOF') === 0) && (message.indexOf('Organization:') !== -1)) {
+                message = _("'Organization' required to register.");
+            } else if ((message.indexOf('EOF') === 0) && (message.indexOf('Username:') !== -1)) {
+                message = _("Login/password or activation key required to register.");
+            } else if (message.indexOf('Must provide --org with activation keys') !== -1) {
+                message = _("'Organization' required when using activation keys.");
+            } else if (message.indexOf('The system has been registered') !== -1) {
+                /*
+                 * Currently we don't separate registration & subscription.
+                 * Our auto-attach may have failed, so close the dialog and
+                 * update status.
+                 */
+                deferred.resolve();
+                return;
+            }
+            requestUpdate();
+            var error = new Error(message);
+            deferred.reject(error);
+        });
+
+    promise = deferred.promise();
+    promise.cancel = function cancel() {
+        process.close("cancelled");
+        // we have no idea what the current state is
+        requestUpdate();
+    };
+
+    return promise;
+};
+
+client.unregisterSystem = function() {
+    var deferred = $.Deferred();
+
+    var args = ['subscription-manager', 'unregister'];
+
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    var process = cockpit.spawn(args, {
+        directory: '/',
+        superuser: "require",
+        environ: ['LC_ALL=C'],
+        err: "out"
+    });
+
+    client.subscriptionStatus.status = "unregistering";
+    needRender();
+    var promise;
+    var buffer = '';
+    process
+        .input('')
+        .stream(function(text) {
+            buffer += text;
+        })
+        .done(function(output) {
+            deferred.resolve();
+        })
+        .fail(function(ex) {
+            if (ex.problem === "cancelled") {
+                deferred.reject(ex);
+                return;
+            }
+            var error = new Error(buffer.trim());
+            deferred.reject(error);
+            requestUpdate();
+        });
+
+    promise = deferred.promise();
+    promise.cancel = function cancel() {
+        process.close("cancelled");
+        // we have no idea what the current state is
+        requestUpdate();
+    };
+
+    return promise;
+};
+
+function statusUpdateFailed() {
+    if (updateTimeout !== null) {
+        client.subscriptionStatus.status = "not-found";
```

Comment:
> Perhaps common code for both this, and handling check_status() failed would be good?

We can log something, sure. I don't think common code would help.
`check_status` is just a dbus method to trigger subscription-manager to send out its current state (asynchronously).
`statusUpdateFailed` is the timeout for said update to appear. That said, I think you're right and the failure handler for sending `check_status` could in turn also call `statusUpdateFailed`